libcloud/compute/drivers/kubevirt.py

```diff
@@ -0,0 +1,996 @@
+            try:
+                result = self.connection.request(req, method=method)
+
+            except Exception as exc:
+                raise
+
+        pv = volume.name
+        req = ROOT_URL + "persistentvolumes/" + pv
+
+        try:
+            result = self.connection.request(req, method=method)
+            return result.status
+        except Exception as exc:
+            raise
+
+    def attach_volume(self, volume, ex_node, **kwargs):
+        """
+        kwargs: bus, name , device (disk or lun)
+        """
+        # volume must be bound to a claim
+        if not volume.extra['isBound']:
+            raise ValueError("""
+            This volume is not bound to a claim,
+            pick a volume that is or create a claim""")
+        claimName = volume.extra['pvc']['name']
+        name = kwargs.get('name', claimName)
+        bus = kwargs.get('bus', 'virtio')
+        device = kwargs.get('device', 'disk')
+        namespace = volume.extra['pvc']['namespace']
+        # check if vm is stopped
+        self.ex_stop_node(ex_node)
+        # check if it is the same namespace
+        if ex_node.extra['namespace'] != namespace:
+            msg = "The PVC and the VM must be in the same namespace"
+            raise ValueError(msg)
+        vm = ex_node.name
+        req = KUBEVIRT_URL + 'namespaces/' + namespace + '/virtualmachines/'\
+            + vm
+        disk_dict = {device: {'bus': bus}, 'name': name}
+        volumes_dict = {'persistentVolumeClaim': {'claimName': claimName},
+                        'name': name}
+        # Get all the volumes of the vm
+        try:
+            result = self.connection.request(req).object
+        except Exception as exc:
+            raise
+        disks = result['spec']['template']['spec']['domain'][
+            'devices']['disks']
+        volumes = result['spec']['template']['spec']['volumes']
+        disks.append(disk_dict)
+        volumes.append(volumes_dict)
+        # now patch the new volumes and disks lists into the resource
+        headers = {"Content-Type": "application/merge-patch+json"}
+        data = {'spec': {
+            'template': {
+                'spec': {
+                    'volumes': volumes,
+                    'domain': {
+                        'devices':
+                        {'disks': disks}
+                    }
+                }
+            }
+        }
+        }
+        try:
+            result = self.connection.request(req, method="PATCH",
+                                             data=json.dumps(data),
+                                             headers=headers)
+            if 'pvcs' in ex_node.extra:
+                ex_node.extra['pvcs'].append(claimName)
+            else:
+                ex_node.extra['pvcs'] = [claimName]
+            return result in VALID_RESPONSE_CODES
+        except Exception as exc:
+            raise
+
+    def detach_volume(self, volume, ex_node):
```

Comment:
Same here - base API method only takes ``volume`` argument. If node argument is also needed, it should be left as as is (prefixed with ``ex_`` since it's indeed not a standard argument).